### ISSUE TITLE
Potential Inclusion of Geant4 Kernel Initialization in Per-Event Timing

### ISSUE THREAD

**Issue by atolosadelgado** [AIDASoft/DD4hep#1364]
### Check duplicate issues.

- [ ] Checked for duplicates

### Goal

TLDR: Proper measure runtime per-event during simulation execution, or clarify that the Time per event include the initialization time.

---

This is an example of the final message of DDSim,
```
DDSim            INFO StartUp Time: 118.18 s, Event Processing: 9.61 s (0.96 s/Event) 
```

I think `Event Processing` includes run initialization time (e.g. physics initialization, voxelization, etc), and the printed time corresponds to the following formula:

```
Run processing time = initialization + Number_of_events * Time_per_event
 ```

I got to this conclusion by comparing a the "event processing time" of 1 and 10 events using a very complex geometry (e.g. CLD). Reproducer and numbers below.

### Operating System and Version

Alma9

### compiler

GCC 14

### ROOT Version

6.32

### DD4hep Version

latest

### Reproducer

The following code run the CLD simulation, completely dummy because the default primary particle (muon) goes along Z axis (default direction) and does not enter any sensitive part of the detector therefore the time per event is short. The comparison between running 1 and 10 events show that the difference of "Time per event is very small", because most of the time in this case is spent in initializing the geometry of CLD. I have added a `/run/verbose 2` in a Geant4 macro file to show more messages from the initialization.

```
git clone --depth 1 git@github.com:key4hep/k4geo.git
cd k4geo/
echo "/run/verbose 2" > verbose.mac
source /cvmfs/sw-nightlies.hsf.org/key4hep/setup.sh 
ddsim --compactFile FCCee/CLD/compact/CLD_o3_v01/CLD_o3_v01.xml --macroFile verbose.mac -G -N 1
ddsim --compactFile FCCee/CLD/compact/CLD_o3_v01/CLD_o3_v01.xml --macroFile verbose.mac -G -N 10
ddsim --compactFile FCCee/CLD/compact/CLD_o3_v01/CLD_o3_v01.xml --macroFile verbose.mac -G -N 100
ddsim --compactFile FCCee/CLD/compact/CLD_o3_v01/CLD_o3_v01.xml --macroFile verbose.mac -G -N 1000
```

I got the following numbers running in my personal laptop
1. One event,
```
DDSim            INFO StartUp Time: 111.89 s, Event Processing: 9.54 s (9.54 s/Event) 
```
2. Ten events:
```
DDSim            INFO StartUp Time: 118.18 s, Event Processing: 9.61 s (0.96 s/Event) 
```
3. Hundred events:
```
DDSim            INFO StartUp Time: 118.73 s, Event Processing: 10.62 s (0.11 s/Event) 
```
4. Thousand events
```
DDSim            INFO StartUp Time: 128.32 s, Event Processing: 18.81 s (0.02 s/Event)
```

The Event processing time has a linear dependency with the number of events, where the equation looks like:
```
Event processing time (s) = 9.54 s + N*0.01s
```

![image](https://github.com/user-attachments/assets/e6bbc7ca-8632-4d8e-bf14-aa2778fdb09d)

**Comment by andresailer** [comment]
We don't really have any way to get a callback from Geant4 about that time it spends to set things up, so we are probably only able to change that message to clarify that it includes some constant time in the beginning.

**Comment by MarkusFrankATcernch** [comment]
We should not invest too much time into this topic. It is really not worth it and besides Alvaro hardly anybody cares....
We should rather investigate issues which are more beneficial to users.
Just adding this for a friend....
